aspnetcore/fundamentals/hosting.md

```diff
@@ -1,225 +1,534 @@
+---
+
+### Environment
 
-### Host Configuration Values
+Sets the app's environment.
 
-**Capture Startup Errors** `bool`
+**Key**: environment  
+**Type**: *string*  
+**Default**: Production  
+**Set using**: `UseEnvironment`
 
-Key: `captureStartupErrors`. Defaults to `false`. When `false`, errors during startup result in the host exiting. When `true`, the host will capture any exceptions from the `Startup` class and attempt to start the server. It will display an error page (generic, or detailed, based on the Detailed Errors setting, below) for every request. Set using the `CaptureStartupErrors` method.
+You can set the *Environment* to any value. Framework-defined values include `Development`, `Staging`, and `Production`. Values aren't case sensitive. By default, the *Environment* is read from the `ASPNETCORE_ENVIRONMENT` environment variable. When using [Visual Studio](https://www.visualstudio.com/), environment variables may be set in the *launchSettings.json* file. For more information, see [Working with Multiple Environments](xref:fundamentals/environments).
 
-Note: When your app runs with Kestrel and IIS, the default behavior is to capture startup errors. 
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
 
 ```csharp
-new WebHostBuilder()
-    .CaptureStartupErrors(true)
-   ```
+WebHost.CreateDefaultBuilder(args)
+    .UseEnvironment("Development")
+    ...
+```
+
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
+
+```csharp
+var host = new WebHostBuilder()
+    .UseEnvironment("Development")
+    ...
+```
+
+---
 
-**Content Root** `string`
+### Hosting Startup Assemblies
 
-Key: `contentRoot`. Defaults to the folder where the application assembly resides (for Kestrel; IIS will use the web project root by default). This setting determines where ASP.NET Core will begin searching for content files, such as MVC Views. Also used as the base path for the [Web Root Setting](#web-root-setting). Set using the `UseContentRoot` method. Path must exist, or host will fail to start.
+Sets the app's hosting startup assemblies.
+
+**Key**: hostingStartupAssemblies  
+**Type**: *string*  
+**Default**: Empty string  
+**Set using**: `UseSetting`
+
+A semicolon-delimited string of hosting startup assemblies to load on startup. This feature is new in ASP.NET Core 2.0.
+
+Although the configuration value defaults to an empty string, the hosting startup assemblies always include the app's assembly. When you provide hosting startup assemblies, they're added to the app's assembly for loading when the app builds its common services during startup.
+
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
 
 ```csharp
-new WebHostBuilder()
-    .UseContentRoot("c:\\mywebsite")
-   ```
+WebHost.CreateDefaultBuilder(args)
+    .UseSetting(WebHostDefaults.HostingStartupAssembliesKey, "assembly1;assembly2")
+    ...
+```
+
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
```

Comment:
This feature is new in 2.0 so is it possible to remove the 1.x tab since it's irrelevant? There are also other instances of this.